### PR TITLE
refac: add plonk namespace

### PR DESCRIPTION
**Pull request by Insun35**
Add `zk::plonk` namespace and set `zk::plonk::halo2` as a nested namespace.

This PR should be merged after #269

### CODE REVIEW

File: tachyon/zk/expressions/expression_type.cc [hunk 1]
```diff
@@ -15,15 +15,15 @@ std::string_view ExpressionTypeToString(ExpressionType type) {
     case ExpressionType::kConstant:
       return "Constant";
     case ExpressionType::kSelector:
-      return "Selector";
+      return "plonk::Selector";
```
Comment (chokobole) [inline tachyon/zk/expressions/expression_type.cc:18]:
```suggestion
      return "Selector";
```

File: tachyon/zk/expressions/expression_type.cc [hunk 2]
```diff
@@ -15,15 +15,15 @@ std::string_view ExpressionTypeToString(ExpressionType type) {
...
     case ExpressionType::kFixed:
       return "Fixed";
     case ExpressionType::kAdvice:
       return "Advice";
     case ExpressionType::kInstance:
       return "Instance";
     case ExpressionType::kChallenge:
-      return "Challenge";
+      return "plonk::Challenge";
```
Comment (chokobole) [inline tachyon/zk/expressions/expression_type.cc:26]:
```suggestion
      return "Challenge";
```

File: tachyon/zk/plonk/keys/verifying_key.h [hunk 3]
```diff
@@ -103,7 +103,7 @@ class VerifyingKey : public Key {
 
   template <typename PCS>
   void SetTranscriptRepresentative(const Entity<PCS>* entity) {
-    halo2::PinnedVerifyingKey<F, C> pinned_verifying_key(entity, *this);
+    plonk::halo2::PinnedVerifyingKey<F, C> pinned_verifying_key(entity, *this);
```
Comment (chokobole) [inline tachyon/zk/plonk/keys/verifying_key.h:106]:
```suggestion
    halo2::PinnedVerifyingKey<F, C> 
```

**Review by chokobole — CHANGES_REQUESTED** [review]

**Review by chokobole — APPROVED** [review]
LGTM

File: tachyon/zk/plonk/permutation/grand_product_argument.h [hunk 4]
```diff
@@ -55,7 +57,7 @@ class GrandProductArgument {
   }
 
  private:
-  FRIEND_TEST(LookupArgumentRunnerTest, ComputePermutationProduct);
+  friend class zk::LookupArgumentRunnerTest_ComputePermutationProduct_Test;
```
Comment (ashjeong) [inline tachyon/zk/plonk/permutation/grand_product_argument.h:60]: What does this new random class do? There are other instances of private `FRIEND_TEST`s that aren't changed like this like in `selector_compressor.h?`

Comment (chokobole) [reply tachyon/zk/plonk/permutation/grand_product_argument.h:60]: This is what `FRIEND_TEST()` does. 

```c++
FRIEND_TEST(zk::LookupArgumentRunnerTest, ComputePermutationProduct);
// equivalent to
// friend class LookupArgumentRunnerTest_ComputePermutationProduct_Test;
```

I don't know the exact reason, but doing only above can't look up the symbol correctly. So we need a forward declaration above, then we end up with a question. If we do this ourselves, do we actually need `FRIEND_TEST()`? This question lead us to just flatten like above ourselves.

**Review by ashjeong — APPROVED** [review]
LGTM

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Pull request merged**